### PR TITLE
Initial encrypted network support

### PR DESCRIPTION
**Pull request by erikwilson**

### CODE REVIEW

**Comment by cjellick** [comment]
https://github.com/rancher/k3s/issues/50

**Comment by yamt** [comment]
why did you enable hairpin mode in this change?

**Comment by erikwilson** [comment]
Hairpin mode was enabled because it is a feature that people have asked for, and it is tested as part of the networking kubernetes-perf-test suite. If there is a good reason to disable it please let me know, as far as I could tell there was not much of a downside for turning it on.

I have also been working on updating the networking tests (https://github.com/erikwilson/kubernetes-perf-tests/tree/master/network/benchmarks/netperf) to use the newest versions of the test utilities (iperf & netperf) and large MTUs. I will probably put in a separate PR with ansible scripts for network benchmarking based off https://github.com/erikwilson/rancher-k3s/tree/encrypted-network-testing/contrib/ansible.

**Comment by yamt** [comment]
no, i don't have any reason to disable (or enable) it.
i was just curious how it's related to the rest of the changes.

**Comment by yamt** [comment]
lgtm

File: pkg/daemons/control/server.go [hunk 1]
```diff
@@ -421,20 +431,43 @@ func WritePasswords(passwdFile string, records [][]string) error {
 	return os.Rename(passwdFile+".tmp", passwdFile)
 }
 
+func genEncryptedNetworkInfo(controlConfig *config.Control, runtime *config.ControlRuntime) error {
```
Comment (galal-hussein) [inline pkg/daemons/control/server.go]: How can this be handled in HA mode? as far as I understand this function will result in creating a IPSECPSK for each master node

**Pull request merged**